src/lib.rs

```diff
@@ -1178,6 +1189,36 @@ where
     }
 }
 
+/// A thread-unsafe version of [`set_logger`].
+///
+/// This function is available on all platforms, even those that do not have
+/// support for atomics that is needed by [`set_logger`].
+///
+/// In almost all cases, [`set_logger`] should be preferred.
+///
+/// # Safety
+///
+/// This function is only safe to call when there is no chance that any of its
```

Comment:
I think the unsafety is more limited than this - it's just forbidden to call this method concurrently, right?

Reply:
It's worded this way because AFAIK any access to `LOGGER` while this function runs would be a data race and thus UB. This includes any of the initialization functions, but also any functions doing any logging.

Reply:
I think logging while this function is running is fine. Because all logging function check `STATE` first and only return an actual logging when its `INITIALIZED`, at which point the logging is set.

Reply:
Ah, I see, that makes sense.